### ISSUE TITLE
intel flavor assembly/asm highlighting broken

### ISSUE THREAD

**Issue by Anteru** [pygments/pygments-migration-test#1030]
(Original issue 1326 created by oblitum on 2017-02-01T16:01:40.420235+00:00)

The following code

```python
import pygments.lexers
import pygments.formatters
formatter_class = pygments.formatters.Terminal256Formatter
formatter = formatter_class(style='algol_nu')
lexer = pygments.lexers.get_lexer_for_filename('.asm')
source = 'mov\tDWORD PTR [rax],edx'
print(pygments.highlight(source, lexer, formatter))
```

Results in

![](https://cloud.githubusercontent.com/assets/1269815/22513871/ad8b1a52-e884-11e6-8e1a-ea79a040d15f.png)

This issue was originally reported at https://github.com/cyrus-and/gdb-dashboard/issues/53.

**Comment by Anteru** [comment]
(Original issue 1326 last updated on 2019-06-23T10:40:39.446810+00:00)

(Issue automaticaly closed due to status in Bitbucket: resolved)

**Comment by Anteru** [comment]
(Original comment by cyrus-and on 2019-06-14T14:05:37.842963+00:00)

This still happens in the latest release. Has this at least been acknowledged as something to be fixed or not?

**Comment by Anteru** [comment]
(Original comment by cyrus-and on 2019-06-14T14:38:39.059426+00:00)

OK the problem is that `pygments.lexers.get_lexer_for_filename('.asm')` returns an instance of \`pygments.lexers.TasmLexer\` instead of \`pygments.lexers.NasmLexer\`. I guess relying on the filename is not the right thing to do since there are multiple ambiguities…

**Comment by Anteru** [comment]
(Original comment by Anteru on 2019-06-22T18:58:42.530018+00:00)

Sounds right. Both Nasm and Tasm register .asm, and hence you’re getting the wrong one :disappointed: I've pushed a change which bumps the priority of the `NasmLexer`  \(so it gets picked by default\), and on top of that, it will check for `PROC` and bail out in that case \(while the `TasmLexer`  will report that it likes `PROC`.\) This should resolve this issue, hopefully.

**Comment by Anteru** [comment]
(Original comment by Anteru on 2019-06-22T19:01:49.521006+00:00)

Fixed in 9556a40aba6d

**Comment by Anteru** [comment]
(Original comment by cyrus-and on 2019-06-23T10:40:39.429476+00:00)

Thank you @{557058:34fc1567-22c0-457a-baf3-b49341033f1c} !
